front/lib/models/assistant.ts

```diff
@@ -0,0 +1,241 @@
+}
+
+AssistantAgentMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    message: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "assistant_agent_message",
+    sequelize: front_sequelize,
+  }
+);
+
+export class AssistantMessage extends Model<
+  InferAttributes<AssistantMessage>,
+  InferCreationAttributes<AssistantMessage>
+> {
+  declare id: number;
+  declare sId: string;
+
+  declare version: number;
+  declare rank: number;
+  declare status: CreationOptional<"visible" | "deleted">;
+
+  declare assistantConversationId: ForeignKey<AssistantConversation["id"]>;
+  declare parentId: ForeignKey<AssistantMessage["id"]> | null;
+  declare assistantUserMessageId: ForeignKey<AssistantUserMessage["id"]> | null;
+  declare assistantAgentMessageId: ForeignKey<
+    AssistantAgentMessage["id"]
+  > | null;
+}
+
+AssistantMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    sId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+    },
+    version: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      defaultValue: 0,
+    },
+    status: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "visible",
+    },
+    rank: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "assistant_message",
+    sequelize: front_sequelize,
+    indexes: [
+      {
+        unique: true,
+        fields: ["version", "assistantConversationId", "rank"],
+      },
+    ],
+    hooks: {
+      // TODO @fontanierh: check if we want to add a Check Constraint (from db.ts ?)
+      beforeValidate: (message) => {
+        if (
+          (message.assistantUserMessageId === null) ===
+          (message.assistantAgentMessageId === null)
+        ) {
+          throw new Error(
+            "Exactly one of assistantUserMessageId, assistantAgentMessageId must be non-null"
+          );
+        }
+      },
+    },
+  }
+);
+
+AssistantMessage.belongsTo(AssistantConversation, {
+  foreignKey: "assistantConversationId",
+  onDelete: "CASCADE",
+});
+AssistantConversation.hasMany(AssistantMessage, {
+  foreignKey: "assistantAgentMessageId",
+});
+AssistantUserMessage.hasOne(AssistantMessage, {
+  foreignKey: "assistantUserMessageId",
+  sourceKey: "id",
+  onDelete: "RESTRICT",
+});
+AssistantMessage.belongsTo(AssistantUserMessage, {
+  foreignKey: "assistantUserMessageId",
+});
+AssistantAgentMessage.hasOne(AssistantMessage, {
+  foreignKey: "assistantAgentMessageId",
+  sourceKey: "id",
+  onDelete: "RESTRICT",
+});
+AssistantMessage.belongsTo(AssistantAgentMessage, {
+  foreignKey: "assistantAgentMessageId",
+});
+AssistantMessage.belongsTo(AssistantMessage, {
+  foreignKey: "parentId",
+  as: "parent",
+});
+AssistantMessage.hasMany(AssistantMessage, {
```

Comment:
lots of magic needs commenting

Reply:
Cleaned up quite a bit of those, should be clear enough w/o comments now I think ? lmk

Reply:
let's do belongsTo foreignKey: parentId